### PR TITLE
Derive geometry eligibility from IFC inheritance graph

### PR DESCRIPTION
**Pull request by louistrue**
## Summary

Refactor geometry pipeline functions to derive entity eligibility from the EXPRESS inheritance graph rather than hand-maintained leaf-level whitelists/blacklists. This fixes silent failures for new IFC4X3 subtypes (e.g., `IfcSolarDevice`) and eliminates the need for code changes when new entity types are added.

## Key Changes

- **New module `schema_helpers.rs`**: Extracted `has_geometry_by_name()` from the auto-generated `schema.rs` (marked "DO NOT EDIT") into a dedicated, maintainable module. This function now:
  - Uses `IfcType::is_subtype_of()` to check inheritance relationships instead of string matching
  - Resolves legacy IFC2x3 / removed-in-IFC4x3 names through the existing `legacy_entities` registry
  - Falls back to substring matching only for reinforcement variants and a minimal set of legacy names
  - Includes comprehensive test coverage for building elements, MEP devices, infrastructure types, and edge cases

- **Refactored `is_simple_geometry_type()`** in `wasm-bindings/src/api/styling.rs`:
  - Replaced hardcoded blacklist with inheritance-based checks
  - Now correctly categorizes any subtype of `IfcEnergyConversionDevice`, `IfcDistributionElement`, etc. as secondary priority without explicit entries
  - Handles legacy entity resolution via `get_legacy_entity_info()`

- **Updated exports**: Removed `has_geometry_by_name` from `generated/mod.rs` re-exports and added `schema_helpers` to the public API in `lib.rs`

## Implementation Details

The new approach leverages the EXPRESS type hierarchy:
- Modern entity names are parsed via `IfcType::from_str()` and checked with `is_subtype_of()`
- Legacy names (not in the modern enum) are resolved through `legacy_entities::get_legacy_entity_info()`, which already carries a `has_geometry` flag
- Unknown types default to safe behavior (no geometry for `has_geometry_by_name`, simple priority for `is_simple_geometry_type`)

This eliminates the regression where `IfcSolarDevice` (inheriting from `IfcEnergyConversionDevice`) was silently excluded because the whitelist matched leaf names directly rather than parent types.

Co-authored with @geronimi73 (PR #585).

https://claude.ai/code/session_01GFZJLYHihhje3a8QDD213H

## Summary by CodeRabbit

* **New Features**
  * IFC4X3 subtypes (e.g., IfcSolarDevice) are now automatically included for rendering.

* **Bug Fixes**
  * Geometry detection improved through inheritance-based classification of IFC types.
  * Enhanced legacy entity name mapping for better compatibility.

### CODE REVIEW

File: rust/core/src/schema_helpers.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+/// in this set.
+fn is_non_geometric_spatial(t: IfcType) -> bool {
+    matches!(
+        t,
+        IfcType::IfcSpatialElement
+            | IfcType::IfcSpatialStructureElement
+            | IfcType::IfcBuilding
+            | IfcType::IfcBuildingStorey
+            | IfcType::IfcFacility
+            | IfcType::IfcFacilityPart
```
[the remark this answers is not in the document]
Comment (louistrue) [reply rust/core/src/schema_helpers.rs]: Fixed in 90672dd. `is_non_geometric_spatial` now blocks anything inheriting from `IfcSpatialElement` (with `IfcSpace`/`IfcSite` as the only exemptions), so `IfcBridge`/`IfcRoad`/`IfcRailway`/`IfcMarineFacility` (under `IfcFacility`), their `*Part` variants (under `IfcFacilityPart`), `IfcSpatialZone`, and any future concrete spatial container all collapse to the same answer without the block-list having to enumerate them. Extended the regression test to cover the IFC4X3 facility subtypes and external spatial elements.

---
_Generated by [Claude Code](https://claude.ai/code)_
